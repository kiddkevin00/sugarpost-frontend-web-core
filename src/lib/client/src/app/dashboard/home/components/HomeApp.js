import BaseComponent from '../../../../common/components/BaseComponent';
import React from 'react';

class HomeApp extends BaseComponent {

  constructor(props) {
    super(props);

    this._bind('_onChange');
    this.state = _getState();
  }

  componentDidMount() {

  }

  componentWillUnmount() {

  }

  render() {
    return (
      <div className="container-fluid">
        <div className="row">
          <div className="jumbotron jumbotron-background" />
        </div>
        <div className="row">
          <div className="container">
            <div className="col-lg-12 text-center">
              <h2>Welcome to The English University!</h2>
              <h3>
                Here you will find everything you need to become fluent in English. English is
                spoken by billions of people and is the international language of business. No
                matter what your field of study or work, understanding and speaking English is
                vital to your success. We provide English classes and private tutoring to help
                you achieve your goals. Our program covers grammar, vocabulary, pronunciation,
                spelling, listening, speaking, reading, and writing.
              </h3>
            </div>
          </div>
        </div>
      </div>
    );
  }

  _onChange() {
    this.setState(_getState());
  }

}

/*
 * A private method. It should only be used by `setState()` and `getInitialState()` to sync up
 * the data in the Flux's store.
 */
function _getState() {
  return {};
}

export default HomeApp;
