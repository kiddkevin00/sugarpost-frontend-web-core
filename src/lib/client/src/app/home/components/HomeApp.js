import homeStore from '../stores/homeStore';
import homeActionCreator from '../actions/homeActionCreator';
import BaseComponent from '../../../common/components/BaseComponent';
import Footer from '../../../common/components/Footer';
import React from 'react';
import { Element } from 'react-scroll';

class HomeApp extends BaseComponent {

  constructor(props) {
    super(props);

    this._bind('_onChange');
    this.state = _getState();
  }

  componentDidMount() {
    homeStore.addChangeListener(this._onChange);
  }

  componentWillReceiveProps(nextProps) {}

  componentWillUnmount() {
    homeStore.removeChangeListener(this._onChange);
  }

  render() {
    return (
      <div id="home-template">
        <header>
          <div className="header-content">
            <div className="header-content-inner">
              <h1>SUGARPOST</h1>
              <hr />
              <p>We are New York City&#39;s premium Dessert Subscription Service!</p>
              <a href="/register/signup" className="btn btn-primary btn-xl page-scroll">
                Sign Up Now
              </a>
              <br />
              <a href="/#about" className="btn btn-circle page-scroll">
                <i className="fa fa-angle-double-down animated" />
              </a>
            </div>
          </div>
        </header>

        <Element name="main">

          <Element name="about">
            <section className="bg-primary" id="about">
              <div className="container">
                <div className="row">
                  <div className="col-lg-8 col-lg-offset-2 text-center">
                    <h2 className="section-heading">Got a sweet tooth? We&#39;ve got the cure!</h2>
                    <hr className="light" />
                    <p className="text-faded">
                      Sugarpost is a New York based online monthly premium subscription service
                      that provides its customers with e-packages of trendy and innovative desserts
                      ranging from frozen treats to baked goods. We have scoured the internet and
                      attained insight from various food bloggers and dessert connoisseurs to
                      deliver our subscribers a selection of unique desserts around New York City!
                    </p>
                    <a href="/#services" className="btn btn-default btn-xl sr-button page-scroll">
                      Get Started
                    </a>
                  </div>
                </div>
              </div>
            </section>
          </Element>

          <Element name="services">
            <section id="services">
              <div className="container">
                <div className="row">
                  <div className="col-lg-12 text-center">
                    <h2 className="section-heading">How It Works</h2>
                    <hr className="primary" />
                  </div>
                </div>
              </div>
              <div className="container">
                <div className="row">
                  <div className="col-lg-3 col-md-6 text-center">
                    <div className="service-box">
                      <i className="fa fa-4x fa-envelope-o text-primary sr-icons" />
                      <h3>Get Codes</h3>
                      <p className="text-muted">
                        After purchasing a monthly subscription, you’ll receive an e-mail with
                        unique redemption codes during the first week of every month.
                      </p>
                    </div>
                  </div>
                  <div className="col-lg-3 col-md-6 text-center">
                    <div className="service-box">
                      <i className="fa fa-4x fa-child text-primary sr-icons" />
                      <h3>Go Venture</h3>
                      <p className="text-muted">
                        Visit all of the locations on your curated list of destination dessert
                        spots. Your e-package will have a little color on each vendor as well.
                      </p>
                    </div>
                  </div>
                  <div className="col-lg-3 col-md-6 text-center">
                    <div className="service-box">
                      <i className="fa fa-4x fa-shopping-basket text-primary sr-icons" />
                      <h3>Claim Desserts</h3>
                      <p className="text-muted">
                        Present your redemption codes at the participating locations to claim your
                        treats. Take a photo, put it on social media, and enjoy.
                      </p>
                    </div>
                  </div>
                  <div className="col-lg-3 col-md-6 text-center">
                    <div className="service-box">
                      <i className="fa fa-4x fa-share-alt text-primary sr-icons" />
                      <h3>Share The Love</h3>
                      <p className="text-muted">
                        Eat up! If you enjoyed your experience, refer a friend for credit towards
                        your next e-package. If enough friends sign up, you’ll even get a month
                        free!
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </Element>

          <Element name="portfolio">
            <section id="portfolio">
              <div className="container">
                <div className="row">
                  <div className="col-lg-12 text-center">
                    <h2 className="section-heading">Feature Products</h2>
                    <hr className="primary" />
                  </div>
                </div>
              </div>
              <div className="container-fluid">
                <div className="row no-gutter popup-gallery">
                  <div className="col-lg-3 col-sm-6">
                    {/* eslint-disable max-len */}
                    <a href="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="portfolio-box">
                      <img src="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="img-responsive" alt="" />
                      <div className="portfolio-box-caption">
                        <div className="portfolio-box-caption-content">
                          <div className="project-category text-faded">
                            Category
                          </div>
                          <div className="project-name">
                            Project Name
                          </div>
                        </div>
                      </div>
                    </a>
                  </div>
                  <div className="col-lg-3 col-sm-6">
                    <a href="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="portfolio-box">
                      <img src="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="img-responsive" alt="" />
                      <div className="portfolio-box-caption">
                        <div className="portfolio-box-caption-content">
                          <div className="project-category text-faded">
                            Category
                          </div>
                          <div className="project-name">
                            Project Name
                          </div>
                        </div>
                      </div>
                    </a>
                  </div>
                  <div className="col-lg-3 col-sm-6">
                    <a href="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="portfolio-box">
                      <img src="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="img-responsive" alt="" />
                      <div className="portfolio-box-caption">
                        <div className="portfolio-box-caption-content">
                          <div className="project-category text-faded">
                            Category
                          </div>
                          <div className="project-name">
                            Project Name
                          </div>
                        </div>
                      </div>
                    </a>
                  </div>
                  <div className="col-lg-3 col-sm-6">
                    <a href="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="portfolio-box">
                      <img src="http://www.musicinmovement.com/blog/wp-content/uploads/2017/01/StockSnap_QFLIGL7A3M.jpg" className="img-responsive" alt="" />
                      {/* eslint-enable max-len */}
                      <div className="portfolio-box-caption">
                        <div className="portfolio-box-caption-content">
                          <div className="project-category text-faded">
                            Category
                          </div>
                          <div className="project-name">
                            Project Name
                          </div>
                        </div>
                      </div>
                    </a>
                  </div>
                </div>
              </div>
            </section>
            <aside className="bg-dark">
              <div className="container text-center">
                <div className="call-to-action">
                  <br />
                  <h2>Can&#39;t Wait?</h2>
                  <a
                    href="/register/signup"
                    className="btn btn-default btn-xl sr-button"
                  >
                    Sign Up Now
                  </a>
                  <br />
                  <br />
                </div>
              </div>
            </aside>
          </Element>

          <section className="no-padding" id="press">
            <div className="container-fluid">
              <div className="row no-gutter popup-gallery">
                <div className="col-lg-12 col-sm-12">
                  <img
                    src="/assets/images/press.png"
                    className="img-responsive"
                    alt=""
                  />
                </div>
              </div>
            </div>
          </section>

          <Element name="contact">
            <section id="contact">
              <div className="container">
                <div className="row">
                  <div className="col-lg-8 col-lg-offset-2 text-center">
                    <h2 className="section-heading">Let&#39;s Get In Touch!</h2>
                    <hr className="primary" />
                    <p>
                      Interested in partnering with us? Inquiries? Questions? Press coverage?
                      That&#39;s great! We look forward to hearing from you! Send us an e-mail and
                      we will get back to you as soon as possible!
                    </p>
                  </div>
                  <div className="col-lg-6 col-lg-offset-3 text-center">
                    <i className="fa fa-envelope-o fa-3x sr-contact" />
                    <p>
                      <a href="mailto:administrator@mysugarpost.com">
                        administrator@mysugarpost.com
                      </a>
                    </p>
                  </div>
                </div>
              </div>
            </section>
          </Element>

        </Element>

        <Footer />
      </div>
    );
  }

  _onChange() {
    this.setState(_getState());
  }

  static _onSubmit(event, email) {
    // Prevents browser's default behavior (page refresh).
    event.preventDefault();

    homeActionCreator.subscribe(email);
  }

  static _onLink(url) {
    const win = window.open(url, '_blank');

    win.focus();
  }

}

/*
 * A private method. It should only be used by `setState()` and `getInitialState()` to sync up
 * the data in the Flux's store.
 */
function _getState() {
  return {};
}

export default HomeApp;
