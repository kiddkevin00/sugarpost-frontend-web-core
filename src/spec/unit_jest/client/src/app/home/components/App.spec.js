import HomeApp, { UnwrappedHomeApp } from '../../../../../../../lib/client/src/app/home/components/App';
import configureStore from '../../../../../../../lib/client/src/common/store/';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Modal } from 'react-bootstrap';
import { Provider } from 'react-redux';
import React from 'react';

test('Home App component should be rendered correctly', () => {
  const component = shallow(
    <UnwrappedHomeApp isModalOpen={ false } dispatchCloseModal={ () => null } />
  );
  const tree = shallowToJson(component);

  expect(tree).toMatchSnapshot();
});

test('Home App component should render correct number of Modal', () => {
  const component = shallow(
    <UnwrappedHomeApp isModalOpen={ true } dispatchCloseModal={ () => null } />
  );

  expect(component.find(Modal).length).toBe(1);
});

test('Home App component can close the initial popup', () => {
  const component = shallow(
    <UnwrappedHomeApp isModalOpen={ true } dispatchCloseModal={ () => null } />
  );

  expect(component.find(Modal).node.props.show).toBe(true);

  component.find('button').simulate('click');

  pending('Seems to have some issue related to async action.');
  expect(component.find(Modal).node.props.show).toBe(false);
});

test('Home App component can close the initial popup - 2', () => {
  const store = configureStore();
  const component = render(
    <Provider store={ store }>
      <HomeApp isModalOpen={ true } dispatchCloseModal={ () => null } />
    </Provider>
  );

  pending('Seems to have some issue related to async action.');
  expect(component.find('#promotion-modal').length).toBe(1);

  store.dispatch({ type: 'HOME.CLOSE_MODAL' });

  expect(component.find('#promotion-modal').length).toBe(0);
});
