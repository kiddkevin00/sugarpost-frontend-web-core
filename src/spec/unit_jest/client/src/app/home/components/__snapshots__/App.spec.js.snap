// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home App component should be rendered correctly 1`] = `
<div
  id="home-template"
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="text-center"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    id="promotion-modal"
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4"
      >
        Try Our Premium Dessert Subscription Today!
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Thumbnail
        alt=""
        bsClass="thumbnail"
        className="text-center"
        src="/assets/images/sugarpost-logo.png"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-12"
          >
            <h2>
              50% OFF
            </h2>
            <h4>
              Use 
              <b>
                U50VRR
              </b>
               Code for Your First Month
            </h4>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-xs-12"
          >
            <p>
              Limited Time Offer
            </p>
          </div>
        </div>
      </Thumbnail>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <a
        className="btn btn-warning btn-block"
        href="/register/signup?referral_code=U50VRR"
        type="button"
      >
        GET YOUR 50% OFF
      </a>
      <button
        className="btn btn-default btn-block"
        onClick={[Function]}
        type="button"
      >
        NO THANKS
      </button>
    </ModalFooter>
  </Modal>
  <header>
    <div
      className="header-content"
    >
      <div
        className="header-content-inner"
      >
        <h1>
          SUGARPOST
        </h1>
        <hr />
        <h3>
          Manhattan’s Premium Dessert Subscription Service
        </h3>
        <p>
          Subscribe and discover trendy desserts to hidden gems with e-packages which will contain vouchers that can claim in-store desserts or
          <b>
            <i>
               Sugarpost exclusives 
            </i>
          </b>
          from
          <b>
            <i>
               different featured vendors every month! 
            </i>
          </b>
        </p>
        <a
          className="btn btn-primary btn-xl page-scroll"
          href="/register/signup"
        >
          Sign Up Now
        </a>
        <br />
        <a
          className="btn btn-circle page-scroll"
          href="/#about"
        >
          <i
            className="fa fa-angle-double-down animated"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="bg-primary"
    id="about"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-8 col-lg-offset-2 text-center"
        >
          <h2
            className="section-heading"
          >
            Got a sweet tooth? We've got the cure!
          </h2>
          <hr
            className="light"
          />
          <p
            className="text-faded"
          >
            We are a New York based online monthly premium subscription service that provides you with e-packages of trendy and innovative desserts. These e-packages have been assembled by our curators who have scoured the internet and attained insight from influential foodies to dessert connoisseurs, providing you with the city's best dessert discovery experience!
          </p>
          <a
            className="btn btn-default btn-xl sr-button page-scroll"
            href="/#services"
          >
            Learn More
          </a>
        </div>
      </div>
    </div>
  </section>
  <section
    id="services"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12 text-center"
        >
          <h2
            className="section-heading"
          >
            How It Works
          </h2>
          <hr
            className="primary"
          />
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-3 col-md-6 text-center"
        >
          <div
            className="service-box"
          >
            <i
              className="fa fa-4x fa-ticket text-primary sr-icons"
            />
            <h3>
              Subscribe
            </h3>
            <p
              className="text-muted"
            >
              After purchasing a monthly subscription, you will receive an email with vouchers during the first day of every month.
            </p>
          </div>
        </div>
        <div
          className="col-lg-3 col-md-6 text-center"
        >
          <div
            className="service-box"
          >
            <i
              className="fa fa-4x fa-map-marker text-primary sr-icons"
            />
            <h3>
              Discover
            </h3>
            <p
              className="text-muted"
            >
              Venture out to all of the locations we've hand-picked for you. We even included a little background on each featured dessert vendor.
            </p>
          </div>
        </div>
        <div
          className="col-lg-3 col-md-6 text-center"
        >
          <div
            className="service-box"
          >
            <i
              className="fa fa-4x fa-birthday-cake text-primary sr-icons"
            />
            <h3>
              Enjoy
            </h3>
            <p
              className="text-muted"
            >
              Present your vouchers at the participating locations to claim your in-store or Sugarpost exclusive desserts - snap it, share it, and enjoy it!
            </p>
          </div>
        </div>
        <div
          className="col-lg-3 col-md-6 text-center"
        >
          <div
            className="service-box"
          >
            <i
              className="fa fa-4x fa-share-alt text-primary sr-icons"
            />
            <h3>
              Share
            </h3>
            <p
              className="text-muted"
            >
              Refer your friends and earn credits. If enough friends sign up, you will get a month free!
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    id="portfolio"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12 text-center"
        >
          <h2
            className="section-heading"
          >
            Upcoming Vendors
          </h2>
          <hr
            className="primary"
          />
        </div>
      </div>
    </div>
    <div
      className="container-fluid"
    >
      <div
        className="row no-gutter popup-gallery"
      >
        <div
          className="col-lg-offset-2 col-lg-8 col-xs-12"
        >
          <img
            alt=""
            className="img-responsive"
            src="/assets/images/june-vendors.png"
          />
        </div>
      </div>
    </div>
  </section>
  <aside
    className="bg-dark"
  >
    <div
      className="container text-center"
    >
      <div
        className="call-to-action"
      >
        <br />
        <h2>
          Subscribe Today
        </h2>
        <a
          className="btn btn-default btn-xl sr-button"
          href="/register/signup"
        >
          Sign Up Now
        </a>
        <br />
        <br />
      </div>
    </div>
  </aside>
  <aside
    className="bg-darker"
  >
    <div
      className="container text-center"
    >
      <div>
        <h2>
          Featured On
        </h2>
        <hr
          className="primary"
        />
        <h4>
          Coming Soon...
        </h4>
      </div>
    </div>
  </aside>
  <section
    id="contact"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-8 col-lg-offset-2 text-center"
        >
          <h2
            className="section-heading"
          >
            Let's Get In Touch!
          </h2>
          <hr
            className="primary"
          />
          <p>
            Interested in partnering with us? Inquiries? Questions? Press coverage? That's great! We look forward to hearing from you! Send us an email and we will get back to you as soon as possible!
          </p>
        </div>
        <div
          className="col-lg-6 col-lg-offset-3 text-center"
        >
          <i
            className="fa fa-envelope-o fa-3x sr-contact"
          />
          <p>
            <a
              href="mailto:administrator@mysugarpost.com"
            >
              administrator@mysugarpost.com
            </a>
          </p>
        </div>
      </div>
    </div>
  </section>
  <Footer />
</div>
`;
